### PR TITLE
Remove the term "whitelist"

### PR DESCRIPTION
**Pull request by tmeschter**
*Note that this fix has been to ship room before and was approved (request #3457); I then discovered I had created the pull request against master instead of stabilization. This is the same change, against stabilization.*

**Bug:** Fixes DevDiv Bug 1185799

**Customer Scenario**
No customer scenario, per se.

There are a number of places where we use the term "whitelist", and one of them (a code comment) was flagged by Policheck. Given that our code is open I've gone ahead and simply fixed them all.

**Fix**
Several internal types were updated with new names, as well as various fields, locals, and a comment.

**Testing**
After the fix, the code still passes all of our unit and integration tests.

@ManishJayaswal This is ready to go to ship room.

### CODE REVIEW

**Comment by srivatsn** [comment]
:+1:

**Comment by ManishJayaswal** [comment]
Approved on 6/15 by ML Shiproom. This was already approved in a PR against master #3457

**Comment by MartinJohns** [comment]
I found the word `whitespace` 345 in the code base. Will you remove that too?

**Comment by sjmcallister** [comment]
I can't even...why is this a thing?

**Comment by ikrastev** [comment]
This is wrong on so many levels. We are programmers, not politicians ffs!

**Comment by crowell** [comment]
at the risk of stirring up a hornets nest, ignorelist is quite possibly one of the worst names for whitelist/blacklist.  Does ignore mean allow, or does ignore mean deny. In this case it seems like you've used ignorelist to mean whitelist, which totally doesn't make sense to me.

I'd propose blocklist and acceptlist or similar if I were looking for a replacement for whitelist/blacklist.

**Comment by indualagarsamy** [comment]
Totally agree with @crowell. IMO, `blocklist` and `acceptlist` is much better replacement. It also specifies the intent much more clearly than calling it `whitelist` or `blacklist`.

**Comment by shiitake** [comment]
Agree with @crowell and @indualagarsamy. Clarity ftw.

**Comment by Sajgoniarz** [comment]
Seriously? It smells like racism bullsh*t for me. Black and whitelist words have a way different etymology. In my language translating whitelist or black list to block or accept list will be nice and long joke.

**Comment by casper** [comment]
Blocklist!? No way. Now that is offensive. My dad used to be a factory worker making blocks. You making fun of blocks now too with your "programming" skills? 

Actually it seems discriminating to "allow", "block", or "ignore" anything. This shit needs to be completely removed from modern day software. Things like firewalls make my blood boil. Discriminating network packets and the free flow of digital data. You programmers. It's inexcusable how you're getting away with this behavior.

**Comment by cubuspl42** [comment]
I can't believe...

**Comment by cubuspl42** [comment]
Instead of fixing the compiler, you should just whitelist `whitelist` in that wonderful "policheck" tool.

**Comment by morsik** [comment]
Why you're so racist and removing things with word `white`? :>

**Comment by Mapiarz** [comment]
I totally agree. Additionally, I think we should remove static typing. We shouldn't be telling objects what they are until they decide for themselves, at runtime. Segregating objects based on **type**? I think I've heard that before...

**Comment by Popeguy** [comment]
I am feeling offended by anyone offended by word such as "blacklist", "master-slave". Please check your privilege and reconsider your priorities.

**Comment by keiosweb** [comment]
great job, now it's time for https://github.com/torvalds/linux/search?p=1&q=blacklist&utf8=%E2%9C%93 please do, I want to see that.

**Comment by qrlpx** [comment]
:+1: by me.

This backlash by angry brogrammers was to be expected. It's like you just wait all day to find something to throw a temper tantrum about. 

Your terminology might have been ok back in the day, but nowadays we actually care about being mindful about each others backgrounds. If someone tells you that the words you're using are not ok, just be nice and don't use them, it's that simple. Our language changes, all the time, deal with it.

**Comment by MrMetric** [comment]
@qripx You're the ones finding something to be angry about. There is no need to change terminology here.

**Comment by khellang** [comment]
Folks... this is a 1 year old PR. Let's just leave it. It's silly, but that's the way it is.

**Comment by lbialy** [comment]
> If someone tells you that the words you're using are not ok, just be nice and don't use them, it's that simple.
**Comply, comply!** 

@quizlx This isn't a change of words that are widely regarded as offensive. This is a blanket ban on words that have no relation to anything remotely offensive to any sane person. Etymology does not hint any racial load neither: https://en.wikipedia.org/wiki/Blacklisting so why is language gutted?

**Comment by casper** [comment]
@quizlx @lbialy 
>  This isn't a change of words that are widely regarded as offensive.

You make exactly the right point. Words like master-slave, whitelisting, blacklisting have been used by programmers for decades. Way before most people here were even born. They were never even remotely intended to be used for anything else than conveying technical information, and getting the job done. No sinister thought or offensive idea even crossed the minds of the people using these terms. They are de-facto standard words used to describe technical computing terms, nothing else.

Now here then comes a group from the so called snowflake generation and they MAKE UP meaning where none was before, and then after they establish what they think is the new meaning they start to demand change.

It is as if I would suddenly start to accuse Github for designing this website with black text on white background. Is Github insinuating something with this color scheme? Black on white? Imagine the shitstorm we could raise demanding that black text on white background has to be banned, because _I_ came up with this offensive meaning for this type of color scheme.

This whole PC trend in programming is an entirely made up fairy tale by some people who cannot find any other way to contribute constructively to society. It's easier to control people by telling them things that need to be done and changed, than to do something constructive yourself. Make up your own reality, and impose it on everyone else.

**Comment by dchem** [comment]
`Policheck` lol

**Comment by houbaron** [comment]
In ancient China, there was a crime called “**Literary inquisition**”. It was designed by government to make people to shut up.After about 300 years, Chinese citizens had revoluted for so many times.But we haven't get rid of it.Today I found the Literary inquisition exist in free countries,too.

**Comment by marblu** [comment]
left-handed wanna say： STOP using Right/LEFT to stand for what is good/the rest

**Comment by EdiWang** [comment]
可以，这很清真

**Comment by quizlx** [comment]
@lbialy
>Etymology does not hint any racial load neither: https://en.wikipedia.org/wiki/Blacklisting

It doesn't need to. 
What matters is how certain words are received, right now, in the present. And you people are doing a swell job politicizing them, good work.

@casper 
>Words like master-slave, whitelisting, blacklisting have been used by programmers for decades. Way before most people here were even born. 

And in the past, it was completly acceptable for people to use the label "n*gro". It had a different meaning back then, even MLK used it. Would you argue that for this reason, it is not offensive today? Go ahead then, use it. Use it and see how people react.

@houbaron 
>In ancient China, there was a crime called “Literary inquisition”. 

Except *duh* Microsoft changed the wording out of their own volition.
Nobody has been hold a gun to the head, nobody has been threatened, nobody has been dogpiled like you're doing right now.... in fact, at this moment, it seems that you people are the only ones trying to police others. How quick the tide turns, huh?

**Comment by houbaron** [comment]
@quizlx I am from China.So my English is poor. I think you may misunderstand me. The "Literary inquisition" is a crime which was designed by gov. People whose words touch the gov's weakness will be throwed into jail. The "Literary inquisition" often happen in some tiny places, like homophonic, spell & double meaning. Nobody like Racial Discrimination. Asian sometimes are victims, too. But the white-list is just a nickname. People who designed such words may be not intentional. In China, there is a kind of website called yellow page. No body think they are making fun of yellow skin Asian. Even in China, yellow represents porn.

**Comment by condret** [comment]
next: remove the master-branch. It implies that there could be a slave-branch. So having a master-branch is a potential sign for supporting slavery.

Seriously, wake up! All this sjw-code-of-conduct-stuff is totally nonsense. Pls spend you time on fixing real bugs instead.

**Comment by Axumata** [comment]
Note how cultural marxists never want to create their own software and communities.Instead they want to police and control software created by others, under the guise of "inclusivity".

**Comment by deeterius** [comment]
I have a counterproposal, for the sake of clarity and better representation for ethnic minorities. Rename whitelist to 'supremepalegoodgoys' and blacklist to 'untermenchafrosemitics'. Will any of the people with commit bits back this proposal ?

Hail Hydra

**Comment by pHr34kY** [comment]
Perhaps `denylist` / `allowlist`? block/accept seems clumsy.

TBH I would have left it as blacklist/whitelist because it's commonly used and understood, and clarity trumps everything when writing shared code.

**Comment by yarwest** [comment]
@pHr34kY why are you even taking this seriously?

**Comment by jycr753** [comment]
Why did this got approved, is a non sense issue... we are programmers not policial class people so we do not need to be sensible to some but hurt people for a word... is a word!!!

**Comment by MasterChief-John-117** [comment]
This change makes no sense simply because of the clarity you're trying to preserve. The commonly accepted term for a list of people who can't use something is called a blacklist. Changing the term destroys the lingo that has been developed, and now people looking at your code from an outsiders view who haven't seen this pull. 
However, I accept your freedom to do this, it's one of the greatest things about America. Additionally, please remove the "master" branch, this implies that it is greater than the other branches, which is very unfair to them. Thank you, a concerned programmer.

**Comment by StanLindsey** [comment]
Github has the word GIT in the name, a British slur. I demand the whole website changes its branding and remove the odd Black & White theme where I do not identify as either of those colours.

**Comment by moopie** [comment]
ITT: people who never touched this project suddenly have some strong words to say in yet another pointless "I AM OFFENDED BY THIS" drama

drama people pls go

**Comment by x89** [comment]
@moopie 👌👌👌👌👌

File: src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs [hunk 1]
```diff
@@ -12,19 +12,19 @@ namespace Microsoft.CodeAnalysis.CompilerServer
 {
     internal static class AnalyzerConsistencyChecker
     {
-        private static readonly ImmutableArray<string> s_defaultWhiteList = ImmutableArray.Create("mscorlib", "System", "Microsoft.CodeAnalysis");
+        private static readonly ImmutableArray<string> s_defaultIgnorableReferenceNames = ImmutableArray.Create("mscorlib", "System", "Microsoft.CodeAnalysis");
```
Comment (IDisposable) [inline src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs]: please change to `s_defaultSafeReferenceNames`

File: src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs [hunk 2]
```diff
@@ -12,19 +12,19 @@ namespace Microsoft.CodeAnalysis.CompilerServer
...
 
-        public static bool Check(string baseDirectory, IEnumerable<CommandLineAnalyzerReference> analyzerReferences, IAnalyzerAssemblyLoader loader, IEnumerable<string> referenceWhiteList = null)
+        public static bool Check(string baseDirectory, IEnumerable<CommandLineAnalyzerReference> analyzerReferences, IAnalyzerAssemblyLoader loader, IEnumerable<string> ignorableReferenceNames = null)
```
Comment (IDisposable) [inline src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs]: please change to `safeReferenceNames`

File: src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs [hunk 3]
```diff
@@ -12,19 +12,19 @@ namespace Microsoft.CodeAnalysis.CompilerServer
...
         {
-            if (referenceWhiteList == null)
+            if (ignorableReferenceNames == null)
```
Comment (IDisposable) [inline src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs]: please change to `safeReferenceNames`

File: src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs [hunk 4]
```diff
@@ -12,19 +12,19 @@ namespace Microsoft.CodeAnalysis.CompilerServer
...
             {
-                referenceWhiteList = s_defaultWhiteList;
+                ignorableReferenceNames = s_defaultIgnorableReferenceNames;
```
Comment (IDisposable) [inline src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs]: please change to `safeReferenceNames`

File: src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs [hunk 5]
```diff
@@ -12,19 +12,19 @@ namespace Microsoft.CodeAnalysis.CompilerServer
...
             }
 
             try
             {
                 CompilerServerLogger.Log("Begin Analyzer Consistency Check");
-                return CheckCore(baseDirectory, analyzerReferences, loader, referenceWhiteList);
+                return CheckCore(baseDirectory, analyzerReferences, loader, ignorableReferenceNames);
```
Comment (IDisposable) [inline src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs]: please change to `safeReferenceNames`

File: src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs [hunk 6]
```diff
@@ -37,7 +37,7 @@ public static bool Check(string baseDirectory, IEnumerable<CommandLineAnalyzerRe
             }
         }
 
-        private static bool CheckCore(string baseDirectory, IEnumerable<CommandLineAnalyzerReference> analyzerReferences, IAnalyzerAssemblyLoader loader, IEnumerable<string> referenceWhiteList)
+        private static bool CheckCore(string baseDirectory, IEnumerable<CommandLineAnalyzerReference> analyzerReferences, IAnalyzerAssemblyLoader loader, IEnumerable<string> ignorableReferenceNames)
```
Comment (IDisposable) [inline src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs]: please change to `safeReferenceNames`

File: src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs [hunk 7]
```diff
@@ -56,14 +56,14 @@ private static bool CheckCore(string baseDirectory, IEnumerable<CommandLineAnaly
...
-            // First, check that the set of references is complete, modulo items in the whitelist.
+            // First, check that the set of references is complete, modulo items in the safe list.
             foreach (var resolvedPath in resolvedPaths)
             {
                 var missingDependencies = AssemblyUtilities.IdentifyMissingDependencies(resolvedPath, resolvedPaths);
 
                 foreach (var missingDependency in missingDependencies)
                 {
-                    if (!referenceWhiteList.Any(name => missingDependency.Name.StartsWith(name)))
+                    if (!ignorableReferenceNames.Any(name => missingDependency.Name.StartsWith(name)))
```
Comment (IDisposable) [inline src/Compilers/Core/VBCSCompiler/AnalyzerConsistencyChecker.cs]: please change to `safeReferenceNames`

**Comment by morsik** [comment]
@moopie: cause software development is not about political corectness. This is not right or wrong. This is just strange what's happened in this commit!

And yeah. We can say we are offended by this, like author can say he's offended by words "Whitelist" and "Blacklist" that were used in IT for years.

File: src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs [hunk 8]
```diff
@@ -17,16 +17,16 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation
     internal sealed class AnalyzerDependencyChecker
     {
         private readonly HashSet<string> _analyzerFilePaths;
-        private readonly List<IAssemblyWhiteList> _assemblyWhiteLists;
+        private readonly List<IIgnorableAssemblyList> _ignorableAssemblyLists;
```
Comment (IDisposable) [inline src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs]: please change to `List<ISkippableAssemblyList> _skippableAssemblyList`

File: src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs [hunk 9]
```diff
@@ -17,16 +17,16 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation
...
         private readonly IBindingRedirectionService _bindingRedirectionService;
 
-        public AnalyzerDependencyChecker(IEnumerable<string> analyzerFilePaths, IEnumerable<IAssemblyWhiteList> assemblyWhiteLists, IBindingRedirectionService bindingRedirectionService = null)
+        public AnalyzerDependencyChecker(IEnumerable<string> analyzerFilePaths, IEnumerable<IIgnorableAssemblyList> ignorableAssemblyLists, IBindingRedirectionService bindingRedirectionService = null)
```
Comment (IDisposable) [inline src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs]: please change to `IEnumerable<ISkippableAssemblyList> skippableAssemblyList`

File: src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs [hunk 10]
```diff
@@ -17,16 +17,16 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation
...
         {
             Debug.Assert(analyzerFilePaths != null);
-            Debug.Assert(assemblyWhiteLists != null);
+            Debug.Assert(ignorableAssemblyLists != null);
```
Comment (IDisposable) [inline src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs]: please change to `skippableAssemblyList`

File: src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs [hunk 11]
```diff
@@ -17,16 +17,16 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation
...
 
             _analyzerFilePaths = new HashSet<string>(analyzerFilePaths, StringComparer.OrdinalIgnoreCase);
-            _assemblyWhiteLists = assemblyWhiteLists.ToList();
+            _ignorableAssemblyLists = ignorableAssemblyLists.ToList();
```
Comment (IDisposable) [inline src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs]: please change to `_skippableAssemblyList = skippableAssemblyList.ToList()`

File: src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs [hunk 12]
```diff
@@ -46,7 +46,7 @@ public AnalyzerDependencyResults Run(CancellationToken cancellationToken = defau
                 }
             }
 
-            _assemblyWhiteLists.Add(new AssemblyIdentityWhiteList(analyzerInfos.Select(info => info.Identity)));
+            _ignorableAssemblyLists.Add(new IgnorableAssemblyIdentityList(analyzerInfos.Select(info => info.Identity)));
```
Comment (IDisposable) [inline src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs]: please change to `_skippableAssemblyList.Add(new SkippableAssemblyIdentityList(analyzerInfos.Select(info => info.Identity)));`

File: src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs [hunk 13]
```diff
@@ -74,7 +74,7 @@ private ImmutableArray<MissingAnalyzerDependency> FindMissingDependencies(List<A
                         ? _bindingRedirectionService.ApplyBindingRedirects(reference)
                         : reference;
 
-                    if (!_assemblyWhiteLists.Any(whiteList => whiteList.Includes(redirectedReference)))
+                    if (!_ignorableAssemblyLists.Any(ignorableAssemblyList => ignorableAssemblyList.Includes(redirectedReference)))
```
Comment (IDisposable) [inline src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs]: please change to `if (!_skippableAssemblyLists.Any(skippableAssemblyList => skippableAssemblyList.Includes(redirectedReference)))`

**Comment by kromit** [comment]
This PR is linked on [/r/ProgrammerHumor](https://www.reddit.com/r/ProgrammerHumor/comments/5u6nxq/when_your_code_is_too_racist/) hence the comments.

**Comment by jonahoffline** [comment]
What about compiling to gender neutral non-binary bits instead?

**Comment by codingducks** [comment]
was thinking quacklist and honklist would be suitable

**Comment by StevenGann** [comment]
Can we replace all Booleans with floats as well? The reasoning is two-fold:

1. Bool and Boolean are named after George Boole, a white, straight, British male who lived in the 19th Century, the height of British and Western Colonialism and the oppression/genocide of people of color, i.e. everyone outside of Europe. Honoring him with a data type, let alone two, is an insult to aboriginal cultures everywhere.

2. Boolean variables are forced to be "True" or "False", implying that anything can be identified as objectively true or objectively false. This is a heteronormative, imperialistic Western philosophy and is the root of all racism, sexism, classism, ageism, speciesism, Fascism, neurotypicalism. Non-European cultures believe that everything is some combination of True AND False, and as a result had no hatred, warfare, discrimination, poverty, or starvation until European Imperialists showed up.

The obvious solution is to replace bools with floats.

(The above is satire, just to be perfectly clear.)

**Comment by huysentruitw** [comment]
@StevenGann PR underway!

**Comment by tekguy** [comment]
This is interesting, where can I find a copy of Policheck?

**Comment by SeriousBusinessCat** [comment]
@tekguy you can find Policheck here https://goo.gl/tqIRl4

**Comment by shiitake** [comment]
@tekguy don't click on that above link! Completely NSFL!

If you do an internet search you can find a presentation on PoliCheck on prezi.com.

File: src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs (hunk 13 above)
Comment (Mikaka27) [inline src/VisualStudio/Core/Def/Implementation/AnalyzerDependencyChecker.cs]: This is stupid, everyone knows what whitelist means, and ignorableAssemblyList means nothing on the first read.

**Comment by manwe-pl** [comment]
That change is just pathetic… Same as people approving it. Fuckin' SJWs.

**Comment by morsik** [comment]
@tekguy: just don't search for that at all. Everyone will have better sleep.

**Comment by tekguy** [comment]
@shiitake ,

Thanks!

**Pull request merged**